### DISCUSSION TITLE
タスクリストのリストマーカーについて

### DISCUSSION THREAD

**Discussion by yarakos95** [increments/qiita-discussions#310] [要望・アイデア・不具合]
現在のタスクリストのリストマーカーは 2 点、奇妙な点があります。

- リストマーカーとリスト項目が近い
- リストマーカーがビュレットとチェックボックス

### リストマーカーとリスト項目が近い

タスクリストのリストマーカーとリスト項目が近いです。どれくらい離すべきかは議論の余地がありますが、現状ではかなり近いと言えると思います。

![image](https://user-images.githubusercontent.com/75878492/194721478-e665d85e-e996-4e2a-beaa-28a595cdd95d.png)

対処療法として、リスト項目とリストマーカーの間のスペースを 2 つにすることで間隔を調整することが出来ますが、これは本来、パーサ側に吸収される部分だと理解しています。

![image](https://user-images.githubusercontent.com/75878492/194722075-631312f8-2e03-4594-92a5-85627acdab66.png)

<details><summary>サンプルコード</summary>

```markdown
- [ ] ここはスペース 1 つ
- [x]   ここはスペース 2 つ
- [x]     ここはスペース 5 つ
```

</details>

GitHub では以下のように表示されます。

```markdown
- [ ] ここはスペース 1 つ
- [x]   ここはスペース 2 つ
- [x]     ここはスペース 5 つ
```

- [ ] ここはスペース 1 つ
- [x]   ここはスペース 2 つ
- [x]     ここはスペース 5 つ

2 つ以上のスペースは無視されています。

### リストマーカーがビュレットとチェックボックス

上の Github での例を見ても明らかなように、タスクリストのリストマーカーがビュレットとチェックボックスとなっていることは奇妙なように思います。チェックボックスのみで十分です。

**Comment by kyntk** [comment]
@Yarakashi-Kikohshi ご報告いただきありがとうございます！
スタイルを調整いたします！

**Comment by kskwtnk** [comment]
@Yarakashi-Kikohshi 

お待たせしました！
タスクリストについて、まずはスタイルの調整を実施しました！

| 変更前 | 変更後 |
| --- | --- |
| ![](https://qiita-image-store.s3.ap-northeast-1.amazonaws.com/0/88/69b086d4-7cfe-cf46-ed15-e4f4ddc5af4e.png) | ![](https://qiita-image-store.s3.ap-northeast-1.amazonaws.com/0/88/1ba96959-7ac5-3aa0-c92c-8dc09aa7febb.png) |

**Reply by yarakos95** [reply]
対応ありがとうございます！
リストマーカーと項目との間が適度に開いており見やすくなりました。また、リストマーカーがチェックボックスのみとなっており、こちらの方が良いと思います。

確認したところ、`[ ]`・`[x]` と項目との間に 2 個以上の半角スペースを挿入すると、項目の位置が右に少しずれる現象（赤線で示した部分）は残されたままになっています。大きな問題にはならないと思いますが、これは仕様として残される部分でしょうか？

![image](https://user-images.githubusercontent.com/75878492/196617698-4d170b05-9a4a-49b2-b011-484a68cb1e9d.png)

**Reply by kskwtnk** [reply]
@Yarakashi-Kikohshi 

すみません、説明が漏れていました＆誤ってクローズのタグをつけていました 🙇 
スペースが空いてしまう現象に対処するのは少し時間がかかりそうだったため、まず目立つ要素としてlist-styleの設定を更新しました

追って対応いたします 🙏

**Reply by yarakos95** [reply]
問題の把握と追って対応の件、了解しました。問題が解決することを期待しています :fire:

**Reply by wataru86** [reply]
@Yarakashi-Kikohshi

お待たせしました 🙇
スペースが空いてしまう不具合を修正しました！

**Reply by yarakos95** [reply]
@wataru86 

対応ありがとうございます！
新規投稿や編集中のプレビューでの修正を確認しました。

### 修正されていない箇所？

[すでに投稿した記事][article] では修正されていないような気がしています。（この記事に問題があるのでしょうか）

[article]: https://qiita.com/Yarakashi_Kikohshi/items/de32784b44e96f5fa1e8#:~:text=%2D%20%5B%20%5D%20%20%20%20%20%E3%81%A9%E3%82%8C%E3%81%A0%E3%81%91%E3%82%A4%E3%83%B3%E3%83%87%E3%83%B3%E3%83%88%E3%81%97%E3%81%A6%E3%82%82%E7%84%A1%E8%A6%96%E3%81%95%E3%82%8C%E3%82%8B

![image](https://user-images.githubusercontent.com/75878492/198037866-d2709e04-104e-4ad4-8c7f-c307763ca8ee.png)

> ⇧ チェックボックスの後ろのスペースを選択している様子

**Reply by wataru86** [reply]
@Yarakashi-Kikohshi

確認ありがとうございます！

ページ表示体験を向上させるため記事のHTMLをキャッシュしているのですが、今回のパーサー修正ではキャッシュの更新をしていないため、過去に投稿された記事にはまだ修正が反映されていません。

投稿済みの記事も一度更新をしていただくと反映されますので、すぐに反映したい場合はお手数ですが記事の編集から更新をお願いします 🙇

**Reply by yarakos95** [reply]
@wataru86 

既存の投稿へ反映されない件、分かりました。そのような設計になっているのですね 👀 

不具合の修正と重ねて教えていただきありがとうございます。
